index.js

```diff
@@ -158,15 +158,16 @@ function monkeypatch() {
     }
   }
 
-  function checkIdentifierOrVisit(node) {
+  function checkIdentifierOrVisit(node, typeParameters) {
     if (node.typeAnnotation) {
       visitTypeAnnotation.call(this, node.typeAnnotation);
     } else if (node.type === "Identifier") {
       // exception for polymorphic types: <T>, <A>, etc
-      if (node.name.length === 1 && node.name === node.name.toUpperCase()) {
-        return;
+      if (node.name.length === 1 && node.name === node.name.toUpperCase() && typeParameters) {
```

Comment:
Not sure why but I was assuming that they were one capital letter given the examples I saw haha.